File: pset6/readability/readability.py
#getting input of text from the user
text=input("Text:")

#counting words, sentences and words in text provided
l,s,w=0,0,1

for i in range(len(text)):
    if text[i].isalpha():
        #counting letters
        l=l+1
    if text[i].isspace():
        #counting words
        w=w+1
    if text[i]=="." or text[i]=="?" or text[i]=="!":
        #counting sentences
        s=s+1

#counting average letters and sentences every 100 words
L=(l*100)/w
S=(s*100)/w

index = round(((0.0588 * L) - (0.296 * S) - 15.8))

#printing reading level
if index<1:
    print("Before Grade 1")
elif index>16:
    print("Grade 16+")
else:
    print("Grade ", index)

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main (void)
{
    string text = get_string("Text:");
    
    // Counting no. of words, letters and sentences in provided text.
    int l = 0, w = 1, s = 0; 
    
    for (int i = 0; i < strlen(text); i++)
    {
        // Counting letters
        if (isalpha(text[i]))
        {
            l++;
        }
        // Counting words
        if (isspace(text[i]))
        {
            w++;
        }
        // Counting sentences
        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            s++;
        }
    }
    
    // Calculating no. of letters and sentences per 100 words.
    float L = ((l*100)/w);
    float S = ((s*100)/w);
    
    // Calculating grade index
    float index = (((0.0588 * L) - (0.296 * S) - 15.8));
    // Rounding to nearest integer
    int inx = index + 0.5;
    // Printing Grades
    if (index<1)
    {
        printf("Before Grade 1\n");
    }
    else if (index>16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", inx);
    }
}